buildSrc/src/main/kotlin/scripts/variants.gradle.kts

```diff
@@ -0,0 +1,46 @@
+package scripts
+
+plugins { id("core.android") }
+
+private enum class BuildTypes(val value: String) {
+    DEBUG("debug"), RELEASE("release")
+}
+
+private enum class ProductFlavors(val value: String) {
+    DEV("dev"), INTERNAL("internal"), PUBLIC("public")
+}
+
+private enum class FlavorDimensions(val value: String) {
+    VERSION("version")
+}
+
+android {
+    buildTypes {
+        getByName(BuildTypes.DEBUG.value) {
+            isMinifyEnabled = false
+            applicationIdSuffix = ".${BuildTypes.DEBUG.value}"
+            isDebuggable = true
+        }
+        getByName(BuildTypes.RELEASE.value) {
+            isMinifyEnabled = true
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+
+    flavorDimensions(FlavorDimensions.VERSION.value)
+    productFlavors {
+        create(ProductFlavors.DEV.value) {
```

Comment:
what if we moved the "creation" to some method in ProductFlavor? and then here we'd just do
```
productFlavors {
    ProductFlavors.Dev.create()
    ProductFlavors.Internal.create()
}
```

same could be done for buildTypes.

I might be thinking too object-oriented :D would that be overengineering?

Reply:
I do not see a reason to do this. We would have 2 objects in any case. I can think about refactoring this a bit by extracting some common logic (if any) and passing parameters for the creation. I would do it in another PR though.